gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -443,9 +443,10 @@ OGRErr OGRFlatGeobufLayer::querySpatialIndex() {
         VSIFree(m_featureOffsets);
         m_featureOffsets = nullptr;
     }
-    auto featureOffsetsSize = static_cast<size_t>(featuresCount) * 8;
+    auto featureOffetsCount = static_cast<size_t>(featuresCount);
```

Comment:
is there a chance that featuresCount could be > std::numeric_limits<size_t> on 32 bit platforms ? (or overflow in the below * 8 multiplication)

Reply:
Hmm yes, both issues are relevant.

The max number of features per spec is 64-bit unsigned but I guess practically limited to to 64-bit unsigned / 8 due the logic you point out on 64-bit arch. I don't see any simple way other than that I make hard checks to require lower number of features than that on read/write. If I do my math correctly it should be 536870912 on 32-bit and 2305843009213690000 on 64-bit so hopefully plenty enough. (Other at this point less known practical limitations notwithstanding)

Reply:
I was more thinking about corrupted files than real world ones

Reply:
Ok. I've added checks but only for what the header provide, in this case it's the number of found indices in the spatial index. If the spatial index is corrupted things can surely go very bad, I'll consider if I can add some sanity checks there too.